### PR TITLE
Allow setting `/testsettings:` file for MSTest runner #421

### PR DESCRIPTION
**Pull request by marcosnz**

### CODE REVIEW

File: src/Cake.Common/Tools/MSTest/MSTestRunner.cs [hunk 1]
```diff
@@ -60,6 +60,19 @@ private ProcessArgumentBuilder GetArguments(FilePath assemblyPath, MSTestSetting
...
             }
 
+            // Got a specific test settings file?
+            if (!string.IsNullOrWhiteSpace(settings.TestSettingsFile))
+            {
+                // Add the file as a property.
+                var testSettingsFile = settings.TestSettingsFile.Trim();
+                if (testSettingsFile.Contains(" "))
+                {
+                    testSettingsFile = testSettingsFile.Quote();
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSTest/MSTestRunner.cs]: Always quote the settings file regardless of whether or not it contains whitespace (this is how we've done it elsewhere).

File: src/Cake.Common/Tools/MSTest/MSTestSettings.cs [hunk 2]
```diff
@@ -19,6 +19,12 @@ public sealed class MSTestSettings : ToolSettings
         public bool NoIsolation { get; set; }
 
         /// <summary>
+        /// Gets or sets the test settings file. If set, it is be passed to MSBuild e.g. /testsettings:local.Testsettings
+        /// </summary>
+        /// <value>The test settings file.</value>
+        public string TestSettingsFile { get; set; }
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSTest/MSTestSettings.cs]: * This should be a `FilePath`.
* Should also be renamed to `TestSettings`

**Comment by gep13** [comment]
@marcosnz I just wanted to follow up with @patriksvensson comments.  Are you in a position to update this PR with the suggestions?

Thanks!

**Comment by gep13** [comment]
FYI this Pull Request is related to https://github.com/cake-build/cake/issues/421

**Comment by devlead** [comment]
@marcosnz planning next release, have you had a chance to look at the feedback given by @patriksvensson?

**Comment by gep13** [comment]
@marcosnz just wanted to follow up again.  Have you had a chance to look at the feedback that was provided?

**Comment by patriksvensson** [comment]
Closing this PR for now.

**Pull request closed without merging**